code/game/machinery/portable_turret.dm

```diff
@@ -1043,33 +998,52 @@ Status: []<BR>"},
 	var/enabled = 1
 	var/lethal = 0
 	var/locked = 1
-	var/control_area //can be area name, path or nothing.
+	var/control_area = null //can be area name, path or nothing.
 	var/ailock = 0 // AI cannot use this
 	req_access = list(access_ai_upload)
+	var/list/obj/machinery/porta_turret/turrets = list()
 
-/obj/machinery/turretid/New()
+/obj/machinery/turretid/New(loc, ndir = 0, built = 0)
 	..()
+	if(built)
+		dir = ndir
+		locked = 0
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)
+		pixel_y = (dir & 3)? (dir ==1 ? -24 : 24) : 0
+	power_change() //Checks power and initial settings
+	return
+
+/obj/machinery/turretid/initialize() //map-placed turrets autolink turrets
+	if(control_area && istext(control_area))
+		for(var/area/A in world)
+			if(A.name && A.name==control_area)

```

Comment:
```A.name == control_area``` is fine, since if ```A.name``` is null, it can't match``` control_area```, as you already check that ```control_area``` was text, and that it exists